src/azul/files.py

```diff
@@ -8,9 +8,45 @@
 import os.path
 import tempfile
 from typing import (
+    Tuple,
     Union,
 )
 
+composite_extensions = [
+    'bam.bai',
+    'csv.gz',
+    'csv.zip',
+    'fastq.gz',
+    'mtx.gz',
+    'mtx.zip',
+    'tar.gz',
+    'tsv.gz',
+    'txt.gz',
+]
```

Comment:
As mentioned in stand-up consider `bz2` and `bzip2` as well. Wherever a .gz is possible we should account for a .bzip2 as well. Not sure about .bz2

And consider comprehending this list for the combinatorics.